src/QuizAtom.tsx

```diff
@@ -1,7 +1,322 @@
+`;
+
+const unansweredLabel = css`
+    cursor: pointer;
+    background-color: ${neutralBackgroundColour};
+    ${commonLabelStyle}
+    background-color: ${neutralBackgroundColour};
+    :hover {
+        background-color: ${darkerNeutralBackgroundColour};
+    }
+`;
+
+const correctChosenLabel = css`
+    background-color: ${correctBackgroundColour};
+    ${commonLabelStyle}
+`;
+
+const correctUnchosenLabel = css`
+    background-color: ${correctUnchosenBackgroundColour};
+    ${commonLabelStyle}
+`;
+
+const incorrectChosenLabel = css`
+    background-color: ${incorrectBackgroundColour};
+    ${commonLabelStyle}
+`;
+
+const incorrectUnchosenLabel = css`
+    background-color: ${neutralBackgroundColour};
+    ${commonLabelStyle}
+`;
+
+const labelStyle = (state: AnswerState) => {
+    switch (state) {
+        case AnswerState.Unanswered:
+            return unansweredLabel;
+        case AnswerState.CorrectChosen:
+            return correctChosenLabel;
+        case AnswerState.CorrectUnchosen:
+            return correctUnchosenLabel;
+        case AnswerState.IncorrectChosen:
+            return incorrectChosenLabel;
+        case AnswerState.IncorrectUnchosen:
+            return incorrectUnchosenLabel;
+        default:
+            return commonLabelStyle;
+    }
+};
+
+const commonTextStyle = css`
+    padding-left: 8px;
+    padding-top: 0;
+    min-height: auto;
+`;
+
+const unchosenText = css`
+    color: ${neutral[7]};
+    ${commonTextStyle};
+`;
+
+const chosenText = css`
+    color: ${neutral[100]};
+    ${commonTextStyle};
+`;
+
+const textStyle = (state: AnswerState) => {
+    switch (state) {
+        case AnswerState.IncorrectChosen:
+            return chosenText;
+        case AnswerState.CorrectChosen:
+            return chosenText;
+        default:
+            return unchosenText;
+    }
+};
+
+const revealTextStyle = (state: AnswerState) => css`
+    ${textStyle(state)}
+    ${textSans.xsmall()}
+`;
+
+const fieldsetStyle = css`
+    margin-bottom: 12px;
+    border: 0px;
+    padding: 0px;
+`;
+
+// Will appear on screen readers
+const visuallyHidden = css`
+    border: 0 !important;
+    clip: rect(0 0 0 0) !important;
+    height: 1px !important;
+    margin: -1px !important;
+    overflow: hidden !important;
+    padding: 0 !important;
+    position: absolute !important;
+    width: 1px !important;
+`;
+
+const iconStyle = css`
+    svg {
+        fill: ${neutral[100]};
+        height: 26px;
+        width: 26px;
+        view-box: '0 0 26px 26px';
+    }
+`;
+
+const fullWidthStyle = css`
+    width: 100%;
+`;
+
+const IconWrapper = (el: JSX.Element) => <div className={iconStyle}>{el}</div>;
+
+export const QuizAtom = ({ id, questions }: QuizAtomType): JSX.Element => (
+    <div data-atom-id={id}>
+        <form>
+            {questions.map((question) => (
+                <Question
+                    key={question.id}
+                    number={questions.indexOf(question) + 1}
```

Comment:
Interesting! `map` does provide index as a second param `map((questions, idx`), which might be a bit easier to reason with? Took me a beat to understand what was going as there's a bit of misdirection going on.